app/person-escort-record/controllers/frameworks.js

```diff
@@ -19,6 +21,48 @@ class FrameworksController extends FormWizardController {
     next()
   }
 
+  reduceResponses(accumulator, { value, value_type: valueType, question }) {
+    const field = question.key
+
+    if (valueType === 'object') {
+      accumulator[field] = value.option
+      accumulator[`${field}--${kebabCase(value.option)}`] = value.details
+    }
+
+    if (valueType === 'collection') {
+      accumulator[field] = value.map(item => item.option)
+      value.forEach(item => {
+        accumulator[`${field}--${kebabCase(item.option)}`] = item.details
+      })
+    }
+
+    if (valueType === 'string' || valueType === 'array') {
+      accumulator[field] = value
+    }
+
+    return accumulator
+  }
```

Comment:
_The following comment is based on personal opinions, so feel free to ignore it._

The intent of this code is to branch out to different values based on `valueType`.
A `switch` statement expresses the same intent but it's easier to scan with the eyes (you can't have expressions in `case`s, but you can in `if`s).

Reply:
Hmm, I tend to avoid `switch` statements as I find them harder to read. I think I went with this approach in a similar fashion of a return early pattern, but in this case there's no need to return early so maybe a switch might be easier to read...

```js
if (valueType === 'object') {
  accumulator[field] = value.option
  accumulator[`${field}--${kebabCase(value.option)}`] = value.details
}

if (valueType === 'collection') {
  accumulator[field] = value.map(item => item.option)
  value.forEach(item => {
    accumulator[`${field}--${kebabCase(item.option)}`] = item.details
  })
}

if (valueType === 'string' || valueType === 'array') {
  accumulator[field] = value
}
```

vs

```js
switch (valueType) {
  case 'object'
    accumulator[field] = value.option
    accumulator[`${field}--${kebabCase(value.option)}`] = value.details
    break
  case 'collection'
    accumulator[field] = value.map(item => item.option)
    value.forEach(item => {
      accumulator[`${field}--${kebabCase(item.option)}`] = item.details
    })
    break
  case 'string'
    accumulator[field] = value
    break
  case 'array'
    accumulator[field] = value
    break
}
```

The downside to the switch in this case is that we also need separate statements for `string` and `array` which set the same value on the accumulator so results in a bit of duplication...

@solidgoldpig @danielepolencic what are your thoughts after seeing them side-by-side?

Reply:
Personally I prefer... object literals

```js
const accumulatorMethods = {
  object: (field, value) => {
    accumulator[field] = value.option
    accumulator[`${field}--${kebabCase(value.option)}`] = value.details
  },
  collection: (field, value) => {
    accumulator[field] = value.map(item => item.option)
    value.forEach(item => {
      accumulator[`${field}--${kebabCase(item.option)}`] = item.details
    })
  },
  string: (field, value) => {
    accumulator[field] = value
  },
  array: (field, value) => {
    accumulator[field] = value
  },
}

accumulatorMethods[valueType]
```

/ducks

Normally I would actually prefer something like that, but this seems overkill here. Unless we want to actually extract and test each of those accumulator bits ;)

Maybe it's just because I drank the Crockford kool-aid, but I don't like switch statements. I don't find the replacement easier to read - but maybe that's just because I'm not used to it.

OTOH, maybe we could do with adding some `elses` to the if statements ¯\_(ツ)_/¯

Reply:
>Maybe it's just because I drank the Crockford kool-aid

😂 

I think I'll leave it as it is for now for speed and ease, unless there are any major hesitations?